Source/TeamMate/ViewModels/PullRequestPickerViewModel.cs

```diff
@@ -58,75 +71,69 @@ public PullRequestQueryReviewStatus ReviewStatus
+            set { SetProperty(ref this.assignedTo, value); }
         }
-        public string SelectedAssignedTo
+
+        public string UICreatedBy
         {
-            get { return this._selectedAssignedTo; }
-            set
-            {
-                this._selectedAssignedTo = value;
-                OnPropertyChanged("SelectedAssignedTo");
-            }
+            get { return this.createdBy; }
+            set { SetProperty(ref this.createdBy, value); }
         }
 
-        public string NewAssignedTo
+        public IEnumerable Project
         {
-            set
-            {
-                if (SelectedAssignedTo != null)
-                {
-                    return;
-                }
-
-                if (!string.IsNullOrEmpty(value))
-                {
-                    this._assignedTo.Add(value);
-                    SelectedAssignedTo = value;
-                }
-            }
+            get { return this._project; }
         }
-
-        public IEnumerable CreatedBy
+        public void AddProject(string projectName)
         {
-            get { return this._createdBy; }
+            this._project.Add(projectName);
         }
-        public string SelectedCreatedBy
+        public string SelectedProject
         {
-            get { return this._selectedCreatedBy; }
+            get { return this._selectedProject; }
             set
             {
-                this._selectedCreatedBy = value;
-                OnPropertyChanged("SelectedCreatedBy");
+                this._selectedProject = value;
+                OnPropertyChanged("SelectedProject");
             }
         }
-
-        public string NewCreatedBy
+        public string NewProject
         {
             set
             {
-                if (SelectedCreatedBy != null)
+                if (SelectedProject != null)
                 {
                     return;
                 }
 
                 if (!string.IsNullOrEmpty(value))
                 {
-                    this._createdBy.Add(value);
-                    SelectedCreatedBy = value;
+                    this._project.Add(value);
+                    SelectedProject = value;
                 }
             }
         }
+
         private void Invalidate()
         {
             if (this.queryInfo != null)
             {
                 this.Name = this.queryInfo.Name;
                 this.ReviewStatus = this.queryInfo.ReviewStatus;
-                this.SelectedAssignedTo = this.queryInfo.AssignedTo;
-                this.SelectedCreatedBy = this.queryInfo.CreatedBy;
+                this.UIAssignedTo = this.queryInfo.UIAssignedTo;
+                this.UICreatedBy = this.queryInfo.UICreatedBy;
+            }
+
+            var projects = this.SettingsService.Settings.Projects;
+            if (projects.Count > 0)
+            {
+                foreach (var project in projects)
+                {
+                    AddProject(project.ProjectName);
```

Comment:
can this not result in dupe projects being added? (as in, when this runs a 2nd/Nth time)

Reply:
Great q. It doesn't because the dialog is created every time and invalidated once per lifecycle. I tried to get this to repro quite a bit before, never had luck :)